recipes/full_finetune_distributed.py

```diff
@@ -787,15 +771,31 @@ def train(self) -> None:
                 # Compute loss
                 # Loss is normalized by default so we multiply by the number of tokens
                 # This way we can normalize by the total number of tokens if we're accumulating gradients
-                running_loss += self._loss_fn(logits, labels) * current_num_tokens
+                current_loss = self._loss_fn(logits, labels) * current_num_tokens
```

Comment:
If there was ever a issue with numerical stability, another option for scaling the loss would be:

```python
if grad_accumulation_step == 0:
	base_num_tokens = current_num_tokens

current_loss = loss_fn(logits, labels) * current_num_tokens / base_num_tokens
```

This might over complicate things but I wanted to leave this here if in the future it turns out a reduced gradient/loss is necessary for smaller dtypes.

Reply:
Agreed, we can take a look at this in the future